must-gather/gather-monitoring-info

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env bash
+
+# copyright 2023.
+#
+# licensed under the apache license, version 2.0 (the "license");
+# you may not use this file except in compliance with the license.
+# you may obtain a copy of the license at
+#
+#     http://www.apache.org/licenses/license-2.0
+#
+# unless required by applicable law or agreed to in writing, software
+# distributed under the license is distributed on an "as is" basis,
+# without warranties or conditions of any kind, either express or implied.
+# see the license for the specific language governing permissions and
+# limitations under the license.
+#
+
+set -eu -o pipefail
+
+SCRIPT_DIR=$(cd -- "$(dirname -- "${BASH_SOURCE[0]}")" &>/dev/null && pwd)
+source "${SCRIPT_DIR}/common"
+
+declare -r BASE_COLLECTION_PATH=$1
+declare -r UWM_NS="openshift-user-workload-monitoring"
+declare -r UWM_INFO_DIR="$BASE_COLLECTION_PATH/uwm-info"
+SA_TOKEN="$(oc create token default)"
+CA_BUNDLE="${UWM_INFO_DIR}/ca-bundle.crt"
+
+get_rules() {
+	log "getting prometheus rules"
+	THANOS_RULER_ROUTE="$(oc get routes -n $UWM_NS thanos-ruler -o jsonpath='{.status.ingress[0].host}')"
```

Comment:
could you move all initialisations that is more complex than an variable assignment to an `init` function?  including "BASE_COLLECTION_PATH" and call that in `main` ?